src/hooks/core/chat/useFetchPostNotificationHook.ts

```diff
@@ -2,29 +2,41 @@
 import AnonymousMessageRepo from '../../../service/repo/anonymousMessageRepo';
 import ChannelList from '../../../database/schema/ChannelListSchema';
 import SignedMessageRepo from '../../../service/repo/signedMessageRepo';
+import useDatabaseQueueHook from '../queue/useDatabaseQueueHook';
 import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
+import StorageUtils from '../../../utils/storage';
 
 const useFetchPostNotificationHook = () => {
   const {localDb, refresh} = useLocalDatabaseHook();
+  const {queue} = useDatabaseQueueHook();
 
   const saveNotifications = async (notifications: any[], fromNotificationAPI: any) => {
     if (!localDb) return;
-    const allPromises: Promise<void>[] = notifications.map((notification) => {
-      const channelList = fromNotificationAPI(notification);
-      return channelList.save(localDb);
+
+    notifications.map((notification) => {
+      queue.addJob({
+        label: 'save post notifications',
+        task: async () => {
+          const channelList = fromNotificationAPI(notification);
+          await channelList.save(localDb);
+        }
+      });
+
+      return null;
     });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `map` function is used here, but the return value is not being utilized. If you're just iterating over an array and not transforming it into a new array, consider using `forEach` instead of `map`. This will make your intention clearer to other developers.

```diff
- notifications.map((notification) => {
+ notifications.forEach((notification) => {
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->